react/src/components/atoms/forms/InputSync/index.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import is from 'is';
+import deepEqual from 'fast-deep-equal';
+import { FormContext } from '../Input/context';
+
+class InputSync extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: {},
+      syncContent: this.syncContent,
+      children: this.props.children,
+      formIds: this.props.formIds
+    };
+  }
+  componentDidMount() {
+    this.checkFormContext();
+  }
+  checkFormContext = () => {
+    if (this.context && this.context.isActive) {
+      if (is.array(this.state.formIds) && !is.array.empty(this.state.formIds)) {
+        this.state.formIds.forEach((id) => {
+          this.linkToForm(id);
+        });
+      }
+    }
+  };
+  linkToForm = (id) => {
+    if (!Object.prototype.hasOwnProperty.call(this.context.value[id], 'syncContent')) {
+      const { value } = this.context;
+      value[id].syncContent = [];
+      value[id].syncContent.push(this.syncContent);
+      this.context.updateState({ value });
```

Comment:
seems like this is updating Form context value? is it able to update the value if so can you provide an example?

Reply:
ok got it this is pushing the syncContent function array to value in formContext so that if can be used to update its own state